gemrb/plugins/INIImporter/INIImporter.h

```diff
@@ -26,67 +26,40 @@
 #include "globals.h"
 
 #include <cstring>
+#include <unordered_map>
 #include <vector>
 
 namespace GemRB {
 
-struct INIPair {
-	std::string Name;
-	std::string Value;
-};
-
-class INITag {
-private:
-	std::vector< INIPair> pairs;
-	std::string TagName;
-public:
-	explicit INITag(std::string Name) : TagName(std::move(Name)) {};
-
-	const std::string& GetTagName() const
-	{
-		return TagName;
-	}
-
-	int GetKeyCount() const
-	{
-		return (int) pairs.size();
-	}
-
-	const std::string& GetKeyNameByIndex(int index) const
-	{
-		return pairs[index].Name;
-	}
-
-	bool AddLine(std::string iniLine);
-	StringView GetKeyAsString(StringView Key, StringView Default) const;
-	int GetKeyAsInt(StringView Key, const int Default) const;
-	float GetKeyAsFloat(StringView Key, const float Default) const;
-	bool GetKeyAsBool(StringView Key, const bool Default) const;
-};
-
 class INIImporter : public DataFileMgr {
 private:
-	std::vector<INITag> tags;
+	std::vector<KeyValueGroup> tags;
```

Comment:
why are we using `vector` instead of `set`?

Reply:
I guess that helps us represeting the order in which sections appear in the INI file, there is some comment inside of the code with edge cases.

Reply:
didn't realize we cared about order (seems odd tho)